cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs

```diff
@@ -12,35 +12,46 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-use crate::imports::*;
+use crate::{imports::*, tests::penpal_emulated_chain::penpal_runtime};
 use asset_hub_westend_runtime::xcm_config::bridging::to_ethereum::DefaultBridgeHubEthereumBaseFee;
-use bridge_hub_westend_runtime::EthereumInboundQueue;
+use bridge_hub_westend_runtime::{
+	bridge_to_ethereum_config::EthereumGatewayAddress, EthereumBeaconClient, EthereumInboundQueue,
+};
 use codec::{Decode, Encode};
-use emulated_integration_tests_common::RESERVABLE_ASSET_ID;
+use emulated_integration_tests_common::{PENPAL_B_ID, RESERVABLE_ASSET_ID};
 use frame_support::pallet_prelude::TypeInfo;
 use hex_literal::hex;
 use rococo_westend_system_emulated_network::asset_hub_westend_emulated_chain::genesis::AssetHubWestendAssetOwner;
-use snowbridge_core::{outbound::OperatingMode, AssetMetadata, TokenIdOf};
+use snowbridge_core::{
+	inbound::InboundQueueFixture, outbound::OperatingMode, AssetMetadata, TokenIdOf,
+};
+use snowbridge_pallet_inbound_queue_fixtures::send_native_eth::make_send_native_eth_message;
 use snowbridge_router_primitives::inbound::{
 	Command, Destination, EthereumLocationsConverterFor, MessageV1, VersionedMessage,
 };
 use sp_core::H256;
 use testnet_parachains_constants::westend::snowbridge::EthereumNetwork;
 use xcm_executor::traits::ConvertLocation;
 
-const INITIAL_FUND: u128 = 5_000_000_000_000;
 pub const CHAIN_ID: u64 = 11155111;
 pub const WETH: [u8; 20] = hex!("87d1f7fdfEe7f651FaBc8bFCB6E086C278b77A7d");
+const INITIAL_FUND: u128 = 5_000_000_000_000;
 const ETHEREUM_DESTINATION_ADDRESS: [u8; 20] = hex!("44a57ee2f2FCcb85FDa2B0B18EBD0D8D2333700e");
 const XCM_FEE: u128 = 100_000_000_000;
+const INSUFFICIENT_XCM_FEE: u128 = 1000;
 const TOKEN_AMOUNT: u128 = 100_000_000_000;
+const TREASURY_ACCOUNT: [u8; 32] =
+	hex!("6d6f646c70792f74727372790000000000000000000000000000000000000000");
 
 #[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
 pub enum ControlCall {
 	#[codec(index = 3)]
 	CreateAgent,
 	#[codec(index = 4)]
-	CreateChannel { mode: OperatingMode },
+	CreateChannel {
+		mode: OperatingMode,
+	},
```

Comment:
May not need `CreateChannel` anymore?

Reply:
Good catch! https://github.com/paritytech/polkadot-sdk/pull/7108/commits/d7dee7f53358ae4b4a4fdfdffb331fc686ef2e17